File: src/Model/GeneralStatement/IfStatement.java
package Model.GeneralStatement;

import Model.Expression.IExpression;
import Model.ProgramState;
import Model.Utils.IDictionary;
import Model.Utils.IStack;

public class IfStatement implements IStatement {
    private IExpression condition;
    private IStatement thenStatement;
    private IStatement elseStatement;

    public IfStatement(IExpression condition, IStatement thenStatement, IStatement elseStatement) {
        this.condition = condition;
        this.thenStatement = thenStatement;
        this.elseStatement = elseStatement;
    }

    @Override
    public ProgramState execute(ProgramState state) {
        IDictionary<String, Integer> symTable = state.getSymTable();
        IStack<IStatement> stack = state.getExecutionStack();
        int value = condition.eval(symTable, state.getHeap());

        if (value != 0)
            stack.push(thenStatement);
        else
            stack.push(elseStatement);

        return state;
    }

    @Override
    public String toString() {
        return "if(" + condition + ") then(" + thenStatement + ") else(" + elseStatement + ");";
    }
}

File: Toy-Language-C#-Intepreter/Toy-Language-CS-Interpreter/Toy-Language-CS-Interpreter/Model/Statement/GeneralStatement/IfStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language_CS_Interpreter.Model.Expression;

namespace Toy_Language_CS_Interpreter.Model.Statement.GeneralStatement
{
    class IfStatement : IStatement
    {
        private IExpression expression;
        private IStatement thenStatement, elseStatement;

        public IfStatement(IExpression expression, IStatement thenStatement, IStatement elseStatement)
        {
            this.expression = expression;
            this.thenStatement = thenStatement;
            this.elseStatement = elseStatement;
        }

        public ProgramState Execute(ProgramState state)
        {
            int result = expression.Evaluate(state.SymbolTable);
            if (result == 0)
                state.ExecutionStack.Push(elseStatement);
            else
                state.ExecutionStack.Push(thenStatement);
            return state;
        }

        public override string ToString()
        {
            return "if(" + expression + ") then " + thenStatement + " else " + elseStatement;
        }
    }
}
